ethereum/rpc/types/block_test.go

```diff
@@ -0,0 +1,62 @@
+package types
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestUnmarshalBlockNumberOrHash(t *testing.T) {
```

Comment:
please refactor to use table-driven unit tests

Reply:
ok